app/models/concerns/hostext/search.rb

```diff
@@ -162,7 +162,10 @@ def search_by_params(key, operator, value)
 
         conditions += " AND " unless conditions.blank? || negate.blank?
         conditions += " NOT(#{negate})" unless negate.blank?
-        {:conditions => conditions}
+        {
+          :joins =>  "INNER JOIN nics ON hosts.id = nics.host_id AND nics.\"primary\" = 't'",
```

Comment:
@dLobatog locally this was the only syntax sqlite liked. If you have another suggestion I'll change this.